Frontend/src/components/Register.js

```diff
@@ -0,0 +1,245 @@
+        .max(30, 'Surname must not exceed 30 characters'),
+    userName: Yup.string()
+        .required('User name is required')
+        .min(10, 'User name must be at least 10 characters')
+        .max(30, 'User name must not exceed 30 characters'),
+    email: Yup.string()
+        .required('Email is required')
+        .email('Email is invalid'),
+    address: Yup.string()
+        .required('Address is required')
+        .min(0, 'Address must be at least 0 characters')
+        .max(30, 'Address must not exceed 30 characters'),
+    password: Yup.string()
+        .required('Password is required')
+        .min(6, 'Password must be at least 6 characters')
+        .max(40, 'Password must not exceed 40 characters'),
+    confirmPassword: Yup.string()
+        .required('Confirm Password is required')
+        .oneOf([Yup.ref('password'), null], 'Confirm Password does not match'),
+    cvuMP: Yup.string()
+        .required('CVU MercadoPago is required')
+        .min(22, 'CVU MercadoPago must be exactly 22 characters')
+        .max(22, 'CVU MercadoPago must be exactly 22 characters'),
+    criptoWallet: Yup.string()
+        .required('The CriptoWallet address is required')
+        .min(8, 'The CriptoWallet address must be exactly 8 characters')
+        .max(8, 'The CriptoWallet address must be exactly 8 characters'),
+    acceptTerms: Yup.bool().oneOf([true], 'Accept Terms is required')
+  });
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors }
+  } = useForm({
+    resolver: yupResolver(validationSchema)
+  });
+
+  const onSubmit = userData => {
+
+      AuthService.register(userData).then(
+          (response) => {
+            setMessage(response.data.message);
+            setSuccessful(true);
+            history.push("/activeCripto")
+            window.location.reload();
+          },
+          (error) => {
+            const resMessage =
+                (error.response &&
+                    error.response.data &&
+                    error.response.data.message) ||
+                error.message ||
+                error.toString();
+
+            setMessage(resMessage);
+            setSuccessful(false);
+          }
+      );
+    };
+
+  return (
+      <div className="register-form">
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <div className="form-group">
+            <label>Name</label>
+            <input
+                name="name"
+                type="text"
+                {...register('name')}
+                className={`form-control ${errors.name ? 'is-invalid' : ''}`}
+            />
+            <div className="invalid-feedback">{errors.name?.message}</div>
+          </div>
+
+          <div className="form-group">
+            <label>Surname</label>
+            <input
+                name="surname"
+                type="text"
+                {...register('surname')}
+                className={`form-control ${errors.surname ? 'is-invalid' : ''}`}
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/JeremiasFuentes/unq-desapp-fuentes-diaz/pullRequest?prid=8198099)
```suggestion
                className={`form-control ${errors.surname ? "is-invalid" : ''}`}
```